### PR TITLE
Space Travelers' Hub

### PR DESCRIPTION
**Pull request by JEFFKAY16**
## Space Travelers' Hub

In this `pull request`, @Qoosim  and I [@JEFFKAY16 ] worked collaboratively and did the following:
- Shared responsibility to work on the project - @Qoosim worked on the `rockets` while I worked on the `missions`. Together we worked on the `profile` section. 
- We used the stated APIs to fetch data for our respective sections
- Made good use of merge conflict resolutions
- Added tests to our application.
- Deployed the app

### CODE REVIEW

**Review by bimbolabuari — CHANGES_REQUESTED** [review]
Hi Team, 

You have done a good job so far 🚀  

## Highlights 🎉 

- Correct Gitflow
- The Rockets section displays a list of all available SpaceX rockets. Users can book each rocket by clicking the reservation button or canceling the previously made booking.

- The Missions section displays a list of current missions along with their brief description and participation status. There is also a button next to each mission that allows users to join the selected mission or leave the mission the user joined earlier.

There are some issues that you still need to work on to go to the next project but you are almost there! 🎉 
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>


## Required Changes ♻️
- [ ] As required, Kindly add more unit testing React Testing Library and Jest.

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

--------


_Reviewed by  @bimbolabuari_

File: src/components/Missions/Mission.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
+/* eslint-disable camelcase */
+/* eslint-disable react/destructuring-assignment */
+/* eslint-disable react/prop-types */
```
[the remark this answers is not in the document]
Comment (JEFFKAY16) [reply src/components/Missions/Mission.js:3]: Done

File: src/components/Missions/Mission.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+        </td>
+        <td>
+          <button
+            type="button"
+            onClick={() => (status === 'booked'
+              ? dispatch(leaveMission(id))
+              : dispatch(joinMission(id)))}
+            className={status === 'booked' ? 'leave-btn' : 'join-btn'}
+          >
+            {status === 'booked' ? 'Leave mission' : 'Join mission'}
```
[the remark this answers is not in the document]
Comment (JEFFKAY16) [reply src/components/Missions/Mission.js:38]: Done

**Comment by JEFFKAY16** [comment]
Hi @bimbolabuari,

Thank you for the review. We have done tests for all components that we have as per requirements. Thank you

**Comment by omar-labana** [comment]
# **Approved ✔️**

Hi TEAM,

Congratulations on completing this project! 👏🏼 🎉

You have correctly implemented all of the requirements📃 correctly and without any errors⚠️ or bugs🪲.

Cheers and good luck in your next milestones! 🍀

Feel free to leave any questions or comments in the PR thread(don't forget to tag me 😄) if something is not 100% clear.

---
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by JEFFKAY16** [comment]
Hi @omar-labana,

Thank you. My dashboard says changes are required, please correct that so that we may be able to submit this project as completed. 

Kind Regards,

Jeff,
Cc: @Qoosim

**Comment by omar-labana** [comment]
Hi @JEFFKAY16 AND @Qoosim, 
I'm sure I approved it, if you can, kindly submit a request to SS and ask for another CRR

**Comment by JEFFKAY16** [comment]
Hi @omar-labana,

Our dashboards are showing `changes requested`, you might have accidentally marked it as incomplete. To save on time, I have submitted for another code review, you may please approve it there.

**Comment by omar-labana** [comment]
@JEFFKAY16 My apologies for this, I think it has already been claimed by another CR.

**Review by EricMbouwe — APPROVED** [review]
### Hi @JEFFKAY16, 

# Status: Approved ✔️
Excellent job on this project. Keep it up 👏
Your project is complete! There is nothing else to say other than... it's time to merge it 💪
Congratulations! 🎉

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**